File: chapter 06/recipe 6.4/DniNetAttributes/DniNetAttributesObj.cs
using System;
using System.Runtime.InteropServices;

namespace DniNetAttributes
{
//	[Guid("406DD97B-8C4F-4fd6-A6D4-6F71103184A3")]
	public interface IComAttributes
	{
		void Foo();
	}

//	[Guid("E4B81256-1727-4a1e-99CF-2ECED97B43B4")]
//	[ProgId("DniNetAttributes.DniNetAttributesModified")]
	[ClassInterface(ClassInterfaceType.None)]
	public class DniNetAttributesObj : IComAttributes
	{
		public void Foo()
		{
		}
	}
}

File: chapter 06/recipe 6.4/DniNetAttributesVB/DniNetAttributesObj.vb
Imports System.Runtime.InteropServices

<Guid("299E5465-6ADC-4336-BB54-1F9BEBB311A2")> _
Public Interface IComAttributes
    Sub Foo()
End Interface

<Guid("7F28D289-E549-4e4d-8822-ADD7B8EFD5FF")> _
<ProgId("DniNetAttributesVB.DniNetAttributesModified")> _
<ClassInterface(ClassInterfaceType.None)> _
Public Class DniNetAttributesObj
    Implements IComAttributes

    Public Sub Foo() Implements IComAttributes.Foo
        'not implemented
    End Sub
End Class
